zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/DataSetInfo.scala

```diff
@@ -13,3 +17,125 @@ case class DataSetInfo(name: String,
+      JsSuccess(new Interval(start.getMillis, end.getMillis))
+    }
+
+    override def writes(interval: Interval): JsValue = {
+      JsObject(List("start" -> Json.toJson(interval.getStart), "end" -> Json.toJson(interval.getEnd)))
+    }
+  }
+
+  implicit def tuple2Writes: Writes[Tuple2[String, String]] = Writes[(String, String)](t => Json.obj("level" -> t._1, "field" -> t._2))
+
+  def parseLevels(levelSeq: Seq[Map[String, String]]): Seq[(String, String)] = {
+    levelSeq.map {
+      levelMap => (levelMap.get("level").getOrElse(""), levelMap.get("field").getOrElse(""))
+    }
+  }
+
+  implicit val fieldFormat: Format[Field] = new Format[Field] {
+    override def reads(json: JsValue): JsResult[Field] = {
+      val name = (json \ "name").as[String]
+      val isOptional = (json \ "isOptional").as[Boolean]
+      DataType.withName((json \ "datatype").as[String]) match {
+        case DataType.Number =>
+          JsSuccess(NumberField(name, isOptional))
+        case DataType.Record => ???
+          //TODO think about Record type later
+        case DataType.Point =>
+          JsSuccess(PointField(name, isOptional))
+        case DataType.Bag =>
+          val innerType = (json \ "innerType").as[String]
+          JsSuccess(BagField(name, DataType.withName(innerType), isOptional))
+        case DataType.Boolean =>
+          JsSuccess(BooleanField(name, isOptional))
+        case DataType.Hierarchy =>
+          val innerType = (json \ "innerType").as[String]
+          val levelSeq = (json \ "levels").as[Seq[Map[String, String]]]
+          JsSuccess(HierarchyField(name, DataType.withName(innerType), parseLevels(levelSeq)))
+        case DataType.Text =>
+          JsSuccess(TextField(name, isOptional))
+        case DataType.String =>
+          JsSuccess(StringField(name, isOptional))
+        case DataType.Time =>
+          JsSuccess(TimeField(name, isOptional))
+        case unknown: DataType.Value => JsError(s"field datatype invalid: $unknown")
+      }
+    }
+
+    override def writes(field: Field): JsValue = {
+      val name = field.name
+      val isOptional = field.isOptional
+      val dataType = field.dataType.toString
+      field match {
+        case record: RecordField => JsNull
+        case bag: BagField => JsObject(List(
+          "name" -> JsString(name),
+          "isOptional" -> JsBoolean(isOptional),
+          "datatype" -> JsString(dataType),
+          "innerType" -> JsString(bag.innerType.toString)))
+        case hierarchy: HierarchyField => JsObject(List(
+          "name" -> JsString(name),
+          "isOptional" -> JsBoolean(isOptional),
+          "datatype" -> JsString(dataType),
+          "innerType" -> JsString(hierarchy.innerType.toString),
+          "levels" -> Json.toJson(hierarchy.levels)))
+        case basicField: Field => JsObject(List(
+          "name" -> JsString(name),
+          "isOptional" -> JsBoolean(isOptional),
+          "datatype" -> JsString(dataType)))
```

Comment:
add the default throws in case we hit the unhandled fields? 
`caes any: Field => throw xxxx `